arch/arc/core/arc_smp.c

```diff
@@ -66,6 +66,32 @@ void arch_start_cpu(int cpu_num, k_thread_stack_t *stack, int sz,
 	}
 }
 
+#ifdef CONFIG_SMP
+static void arc_slave_irq_init(void)
+{
+	int irq; /* the interrupt index */
+
+	/* Interrupts from 0 to 15 are exceptions and they are ignored
+	 * by IRQ auxiliary registers. For that reason we skip those
+	 * values in this loop.
+	 */
+	for (irq = 16; irq < CONFIG_NUM_IRQS; irq++) {
+
+		z_arc_v2_aux_reg_write(_ARC_V2_IRQ_SELECT, irq);
+#ifdef CONFIG_ARC_SECURE_FIRMWARE
+		z_arc_v2_aux_reg_write(_ARC_V2_IRQ_PRIORITY,
+			 (CONFIG_NUM_IRQ_PRIO_LEVELS-1) |
+			 _ARC_V2_IRQ_PRIORITY_SECURE); /* lowest priority */
+#else
+		z_arc_v2_aux_reg_write(_ARC_V2_IRQ_PRIORITY,
+			 (CONFIG_NUM_IRQ_PRIO_LEVELS-1)); /* lowest priority */
+#endif
+		z_arc_v2_aux_reg_write(_ARC_V2_IRQ_ENABLE, _ARC_V2_INT_DISABLE);
+		z_arc_v2_aux_reg_write(_ARC_V2_IRQ_TRIGGER, _ARC_V2_INT_LEVEL);
+	}
+}
```

Comment:
Isn't it the same code as we have in ARCv2 interrupt_controller driver?
https://github.com/zephyrproject-rtos/zephyr/blob/380528676960d0fb57fae48d8616154a5e06705c/drivers/interrupt_controller/intc_arcv2_irq_unit.c#L63-L88

If so, we need to reuse it.

Reply:
indeed, I just copy from here, because these codes can't be called in slave cores. 

during master initialization, these codes will be called through:
https://github.com/zephyrproject-rtos/zephyr/blob/380528676960d0fb57fae48d8616154a5e06705c/drivers/interrupt_controller/intc_arcv2_irq_unit.c#L220-L226